src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,18 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                if (false !== strpos($_ENV['PATH'], 'heroku')) {
```

Comment:
Maybe use this detection instead: https://github.com/jedisct1/libsodium-php/blob/e529d49b4ab0a875333eff867722207de97bcb76/tests/pwhash_argon2i.phpt#L5 ?